dali/operators/imgcodec/image_decoder.h

```diff
@@ -751,6 +749,23 @@ class ImageDecoder : public StatelessOperator<Backend> {
       }
     }
 
+    // This is a workaround for nvImageCodec <= 0.2
+    auto any_need_processing = [this]() {
+      for (auto &st : state_) {
+        assert(ws.stream() == st->image_info.cuda_stream);  // assuming this is true
+        if (st->need_processing)
+          return true;
+      }
+      return false;
+    };
+    if (need_host_sync_alloc() && any_need_processing()) {
+      DomainTimeRange tr("alloc sync", DomainTimeRange::kOrange);
+      auto event = CUDAEventPool::instance().Get();
+      CUDA_CALL(cudaEventRecord(event, ws.stream()));
+      CUDA_CALL(cudaEventSynchronize(event));
+      CUDAEventPool::instance().Put(std::move(event));
```

Comment:
This is just `cudaStreamSyncrhonize(ws.stream())`, but with more overhead.

Reply:
Replaced it.